src/m365/pa/commands/app/app-export.ts

```diff
@@ -0,0 +1,238 @@
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async getPackageResources(args: CommandArgs, logger: Logger): Promise<any> {
+    if (this.verbose) {
+      logger.logToStderr('Getting the Microsoft Power App resources...');
+    }
+    const requestOptions: CliRequestOptions = {
+      url: `${this.resource}/providers/Microsoft.BusinessAppPlatform/environments/${formatting.encodeQueryParameter(args.options.environment)}/listPackageResources?api-version=2016-11-01`,
+      headers: {
+        accept: 'application/json'
+      },
+      data: {
+        baseResourceIds: [
+          `/providers/Microsoft.PowerApps/apps/${args.options.id}`
+        ]
+      },
+      responseType: 'json'
+    };
+
+    const response = await request.post<any>(requestOptions);
+    Object.keys(response.resources).forEach((key) => {
+      response.resources[key].suggestedCreationType = 'Update';
+    });
+    return response.resources;
+  }
+
+  private async exportPackage(args: CommandArgs, logger: Logger): Promise<string> {
+    if (this.verbose) {
+      logger.logToStderr(`Initiating package export for Microsoft Power App ${args.options.id}...`);
+    }
+
+    const resources = await this.getPackageResources(args, logger);
+
+    const requestOptions: CliRequestOptions = {
+      url: `${this.resource}/providers/Microsoft.BusinessAppPlatform/environments/${formatting.encodeQueryParameter(args.options.environment)}/exportPackage?api-version=2016-11-01`,
+      headers: {
+        accept: 'application/json'
+      },
+      responseType: 'json',
+      data: {
+        includedResourceIds: [
+          `/providers/Microsoft.PowerApps/apps/${args.options.id}`
+        ],
+        details: {
+          creator: args.options.packageCreatedBy,
+          description: args.options.packageDescription,
+          displayName: args.options.packageDisplayName,
+          sourceEnvironment: args.options.packageSourceEnvironment
+        },
+        resources: resources
+      },
+      fullResponse: true
+    };
+
+    const response: any = await request.post<any>(requestOptions);
+
+    const location = response.headers.location;
+
+    if (this.verbose) {
+      logger.logToStderr(`Location of the Power Apps export: ${location}`);
+    }
+
+    return response.headers.location;
+  }
+
+  private async getPackageLink(args: CommandArgs, logger: Logger): Promise<string> {
+    let status;
+    let link;
+
+    const location = await this.exportPackage(args, logger);
+
+    const requestOptions: CliRequestOptions = {
+      url: location,
+      headers: {
+        accept: 'application/json'
+      },
+      responseType: 'json'
+    };
+
+    do {
+      const response = await request.get<any>(requestOptions);
+      status = response.properties.status;
+      if (status === "Succeeded") {
+        link = response.properties.packageLink.value;
+      }
+      else {
+        await this.sleep(PaAppExportCommand.sleepInterval);
```

Comment:
I don't see the value of using the sleep timeout as a private static variable. Let's keep it simple and put it in the code right where it's needed.

Reply:
I looked into the existing command and saw that it is used as a private static in the `spo site rename` command